### DISCUSSION TITLE
[Feature Request] DateRangePicker isDateUnavailable function should know whether we're selecting the start or end of a range

### DISCUSSION THREAD

**Discussion by corwinstephen** [heroui-inc/heroui#4712] [Feature requests]
### Is your feature request related to a problem? Please describe.

I'm implementing a booking platform and currently I'm not able to distinguish between a day that is completely unavailable and one which is only unavailable for checkin/out. For example, if I have a guest checking in a certain day, then that day is available as a check OUT day, but not as a check IN day, so I need it to be available to select as the end date, but not the start date. Since the function doesn't have access to that info, I can't currently do this.

### Describe the solution you'd like

As a user, I expected that when I click the datepicker, dates that are available for checkin will show as available, and then after I select one, only dates that are available for checkOUT will show as available.

### Describe alternatives you've considered

MUI's daterangepicker has this functionality, but I don't want to ship an entire second library just for this.

**Comment by linear** [comment]
<p><a href="https://linear.app/nextui-inc/issue/ENG-1768/[feature-request]-daterangepicker-isdateunavailable-function-should">ENG-1768 [Feature Request] DateRangePicker isDateUnavailable function should know whether we're selecting the start or end of a range</a></p>

**Comment by mehm8128** [comment]
It seems that `isDateUnavailable` depends on React Aria `useRangeCalendar`: https://github.com/adobe/react-spectrum/blob/50c7ada5d1880a174b6b6d3f43e8d90ee9bd4ad8/packages/%40react-types/calendar/src/index.d.ts#L29-L30

And the similar issue is discussed on here: https://github.com/adobe/react-spectrum/issues/3311
